machxo2/arch.cc

```diff
@@ -532,6 +631,337 @@ DecalXY Arch::getGroupDecal(GroupId group) const
+            (fromPort == id_C0 && toPort == id_WADO2) || (fromPort == id_C1 && toPort == id_WDO2) ||
+            (fromPort == id_D0 && toPort == id_WADO3) || (fromPort == id_D1 && toPort == id_WDO3)) {
+            delay = DelayQuad(0);
+            return true;
+        }
+        return false;
+    } else if (cell->type == id_DCCA) {
+        if (fromPort == id_CLKI && toPort == id_CLKO) {
+            delay = DelayQuad(0);
+            return true;
+        }
+        return false;
+    } else if (cell->type == id_DP8KC) {
+        return false;
+/*    } else if (cell->type.in(id_IOLOGIC, id_SIOLOGIC)) {
+        return false;*/
+    } else {
+        return false;
+    }
+}
+
+TimingPortClass Arch::getPortTimingClass(const CellInfo *cell, IdString port, int &clockInfoCount) const
+{
+    auto disconnected = [cell](IdString p) { return !cell->ports.count(p) || cell->ports.at(p).net == nullptr; };
+    clockInfoCount = 0;
+    if (cell->type == id_TRELLIS_COMB) {
+        if (port == id_WCK)
+            return TMG_CLOCK_INPUT;
+        if (port.in(id_A, id_B, id_C, id_D, id_FCI, id_FXA, id_FXB, id_F1))
+            return TMG_COMB_INPUT;
+        if (port == id_F && disconnected(id_A) && disconnected(id_B) && disconnected(id_C) && disconnected(id_D) &&
+            disconnected(id_FCI))
+            return TMG_IGNORE; // LUT with no inputs is a constant
+        if (port.in(id_F, id_FCO, id_OFX))
+            return TMG_COMB_OUTPUT;
+        if (port == id_M)
+            return TMG_COMB_INPUT;
+        if (port.in(id_WD, id_WAD0, id_WAD1, id_WAD2, id_WAD3, id_WRE)) {
+            clockInfoCount = 1;
+            return TMG_REGISTER_INPUT;
+        }
+        return TMG_IGNORE;
+    } else if (cell->type == id_TRELLIS_FF) {
+        bool using_m = (cell->ffInfo.flags & ArchCellInfo::FF_M_USED);
+        if (port == id_CLK)
+            return TMG_CLOCK_INPUT;
+        if (port == id_DI || (using_m && (port == id_M)) || port.in(id_CE, id_LSR)) {
+            clockInfoCount = 1;
+            return TMG_REGISTER_INPUT;
+        }
+        if (port == id_Q) {
+            clockInfoCount = 1;
+            return TMG_REGISTER_OUTPUT;
+        }
+        return TMG_IGNORE;
+    } else if (cell->type == id_TRELLIS_RAMW) {
+        if (port.in(id_A0, id_A1, id_B0, id_B1, id_C0, id_C1, id_D0, id_D1))
+            return TMG_COMB_INPUT;
+        if (port.in(id_WDO0, id_WDO1, id_WDO2, id_WDO3, id_WADO0, id_WADO1, id_WADO2, id_WADO3))
+            return TMG_COMB_OUTPUT;
+        return TMG_IGNORE;
+    } else if (cell->type == id_TRELLIS_IO) {
+        if (port.in(id_T, id_I))
+            return TMG_ENDPOINT;
+        if (port == id_O)
+            return TMG_STARTPOINT;
+        return TMG_IGNORE;
+    } else if (cell->type == id_DCCA) {
+        if (port == id_CLKI)
+            return TMG_COMB_INPUT;
+        if (port == id_CLKO)
+            return TMG_COMB_OUTPUT;
+        return TMG_IGNORE;
+/*    } else if (cell->type == id_DCSC) {
```

Comment:
can you remove the commented out sections here?

Reply:
this is now removed